os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -149,20 +149,47 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo, size_t textsize, size
 	/* Allocate memory to hold the ELF image */
 
 #ifdef CONFIG_APP_BINARY_SEPARATION
-#ifdef CONFIG_DEBUG_MM_HEAPINFO
-	ARCH_GET_RET_ADDRESS
-	loadinfo->textalloc = (uintptr_t)mm_malloc(loadinfo->uheap, textsize + datasize, retaddr);
-#else
-	loadinfo->textalloc = (uintptr_t)mm_malloc(loadinfo->uheap, textsize + datasize);
-#endif
+        /* Allocate the RAM partition to load the app into */
+        loadinfo->binp->ramsize += loadinfo->textsize + loadinfo->rosize + loadinfo->datasize;
+
+        if (mm_allocate_ram_partition(&loadinfo->binp->ramstart, &loadinfo->binp->ramsize) < 0) {
+                berr("ERROR: Failed to allocate RAM partition\n");
+                return -ENOMEM;
+        }
+
+	if (loadinfo->textsize > loadinfo->rosize) {
+		unsigned long align_mask = loadinfo->textsize - 1;
```

Comment:
This pattern for getting size-aligned value is used in many times, even in os/mm/mm_heap/mm_partition_mgr.c
How about making function for readability?